jerry-core/ecma/operations/ecma-jobqueue.c

```diff
@@ -232,34 +232,39 @@ ecma_process_promise_resolve_thenable_job (void *obj_p) /**< the job to be opera
 {
   ecma_job_promise_resolve_thenable_t *job_p = (ecma_job_promise_resolve_thenable_t *) obj_p;
   ecma_object_t *promise_p = ecma_get_object_from_value (job_p->promise);
+  ecma_promise_resolving_functions_t *funcs = ecma_promise_create_resolving_functions (promise_p);
   ecma_string_t str_resolve, str_reject;
   ecma_init_ecma_magic_string (&str_resolve, LIT_INTERNAL_MAGIC_STRING_RESOLVE_FUNCTION);
   ecma_init_ecma_magic_string (&str_reject, LIT_INTERNAL_MAGIC_STRING_REJECT_FUNCTION);
-
-  ecma_value_t resolve = ecma_op_object_get (promise_p, &str_resolve);
-  ecma_value_t reject = ecma_op_object_get (promise_p, &str_reject);
-
-  ecma_value_t argv[] = { resolve, reject };
+  ecma_op_object_put (promise_p,
+                      &str_resolve,
+                      funcs->resolve,
+                      false);
+  ecma_op_object_put (promise_p,
+                      &str_reject,
+                      funcs->reject,
+                      false);
+
+  ecma_value_t argv[] = { funcs->resolve, funcs->reject };
   ecma_value_t ret;
   ecma_value_t then_call_result = ecma_op_function_call (ecma_get_object_from_value (job_p->then),
                                                          job_p->thenable,
                                                          argv,
                                                          2);
 
+  ecma_promise_free_resolving_functions (funcs);
```

Comment:
isn't it too early to free `funcs` here? `funcs->reject` is still used later

Reply:
Oh, sorry for that